apps/heft/src/schemas/typescript.schema.json

```diff
@@ -51,6 +51,14 @@
     "maxWriteParallelism": {
       "description": "Set this to change the maximum write parallelism. The default is 50.",
       "type": "number"
+    },
+
+    "subprocessNodeArgv": {
+      "description": "Adds extra commandline arguments to pass to the Node subprocess running Typescript. e.g. --max-old-space-size. Supported command line interface for node.js can be found at https://nodejs.org/api/cli.html",
+      "type": "array",
+      "items": {
+        "type": "string"
+      }
```

Comment:
I think it'll be better to specifically call out the options we want to support (perhaps with better terminology than things like `max-old-space-size`).

Reply:
Specifically, the docs could give more context about the intended purpose:
- Why would someone need to specify `max-old-space-size`?
- Why would they specify it for TypeScript only? Why not ESLint or Webpack or whatever?
- Why is it specified via a config file (versus a CLI parameter or environment variable)?

I'm not debating your design, just asking to explain the thought process behind it.

Philosophically Rush Stack is about:
- identifying important problems
- providing a ready-made solution
- explaining how to use the solution (and some background about why it works)
- providing support (e.g. meaningful error messages) if it doesn't work as intended

I think Ian's point is that these goals aren't met very well with the typical flexible solutions (*"and here's a bag where you can stuff some strings for another tool to process"*).

Reply:
@octogonz @iclanton Thanks for explaining your concerns, it's been good food for thought especially the philosophy. In light of this and the issues I'm now running into with porting the test command, I do think it makes more sense to change the design to handle specifically the --max-old-space-size parameter, and for the parent process, which would also be inherited by subprocesses in the current implementation. Does this sound like it would be better as a CLI parameter, or a config file parameter? Does that design shift make sense?

Reply:
The `--max-old-space-size`  parameter is pretty much the only Node.js CLI setting that I've seen being widely used. (But over time we probably will encounter others, e.g. `--trace-warnings` might be useful.) But it probably does makes sense to convert it to be a dedicated option with documentation. It might also allows us to implement it more intelligently.  

If I remember correctly (you might want to double-check this):

- If `--max-old-space-size` is too large, then Node.js will eat up the machine's RAM and will contest with other concurrent processes.  So it is loosely related to the hardware capabilities of a particular machine, however:
- It is mainly used for two specific cases: (1) a project with an unusually large number of input files, or (2) as a temporary workaround for scripts/tests with a known memory leak. This suggests maybe it would be better as a config file setting than an environment variable.
- These two problems tend to affect a specific build task (e.g. Jest, or Webpack, or deployment) so we might want to fine-tune which Heft subprocess gets adjusted.  So maybe we could even have separate config file settings for each task.
- In a pinch, Node.js already provides an environment variable [NODE_OPTIONS](https://nodejs.org/api/cli.html).

So maybe it should be a config file setting instead of a CLI option or environment variable.  Something like this:
```js
/**
 * Configures the maximum size of the process memory pool, measured in megabytes.  
 * This is equivalent to the Node.js "--max-old-space-size" command-line option. Increasing
 * this limit can eliminate "out of memory" errors for a project with unusually large inputs
 * or a build task with a known memory leak.  However, be aware that specifying a value that
 * is too large may impact operating system performance.  Specify 0 to request the Node.js
 * internal default value, as determined by the Node.js release.
 */
// "nodeMemoryLimitMB": 0
```

Maybe we can give meaningful names for the Node.js processes that Heft creates, so users can configure them individually.  It could be like this:

```js
"nodeProcessSettings": [
  {
    /**
     * (Required) Each Node.js process created by Heft is given a process class name indicating
     * the kind of work performed.  The class name can be determined from the
     * "HEFT_PROCESS_CLASS" environment variable for that process.  It also appears in
     * brackets in the Node.js "--title" parameter, for example "Heft [jest]" indicates the
     * "jest" process class.
     *
     * The "nodeProcessSettings" setting determines which process class name this settings block
     * applies to.  The name "default" is a reserved word for the setting block that applies to any
     * process class that does not have an explicit "nodeProcessSettings" entry.
     */
    "nodeProcessClass": "default",

    "nodeMemoryLimitMB": 0
  }
]
```

@iclanton I'm not sure what files these settings would go in, however.  Seems like we need a `heft.json` file.

Also, should there be a way to configure Heft's startup process?  I remember that Gulp did this by reinvoking itself to make a new Node.js process with the requested settings.  That feels a little clunky, but it could be acceptable if it could be performed by a tiny shim with negligible startup cost.

Reply:
Thank you @octogonz for all the feedback! I agree that this would be a good documented feature going forward. I've realized that this feature isn't necessary for my needs if I do configure the heft startup process, which is possible calling heft via the `node` CLI, and for the sake of implementing our actions in heft I've decided to just do that, pass the parameter and it will be inherited by the subprocesses. This is a good spark for a future implementation, though. I'll be pivoting this branch just to support the getConfigHash change and then go from there